tests/scripts/analyze_outcomes.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+"""Analyze the test outcomes from a full CI run.
+
+This script can also run on outcomes from a partial run, but the results are
+less likely to be useful.
+"""
+
+import argparse
+import sys
+import traceback
+
+class Results:
+    """Process analysis results."""
+
+    def __init__(self):
+        self.error_count = 0
+        self.warning_count = 0
+
+    @staticmethod
+    def log(fmt, *args, **kwargs):
+        sys.stderr.write((fmt + '\n').format(*args, **kwargs))
+
+    def error(self, fmt, *args, **kwargs):
+        self.log('Error: ' + fmt, *args, **kwargs)
+        self.error_count += 1
+
+    def warning(self, fmt, *args, **kwargs):
+        self.log('Warning: ' + fmt, *args, **kwargs)
+        self.warning_count += 1
+
+class TestCaseOutcomes:
+    """The outcomes of one test case across many configurations."""
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self):
+        self.successes = []
```

Comment:
I think the contents of those lists should be documented.

Reply:
I still think it would be good to document that items in the list will be strings with format 'platform;config'.

Reply:
Sure, done. I was just waiting to update until I'd understood your other comment.

Reply:
And pushed to the branch of this PR (which I accidentally made from `analyze_outcomes-count_test_cases-1` rather than `analyze_outcomes-count_test_cases`).